exec/java-exec/src/main/java/org/apache/drill/exec/store/schedule/AssignmentCreator.java

```diff
@@ -146,6 +153,32 @@ private AssignmentCreator(List<DrillbitEndpoint> incomingEndpoints, List<T> unit
     return currentUnassignedList;
   }
 
+
+  private LinkedList<WorkEndpointListPair<T>> assignLocal(List<WorkEndpointListPair<T>> workList,
+                                                          Map<DrillbitEndpoint,FragIteratorWrapper> endpointIterators) {
+    LinkedList<WorkEndpointListPair<T>> currentUnassignedList = Lists.newLinkedList();
+    for (WorkEndpointListPair<T> workPair : workList) {
+      DrillbitEndpoint endpoint = workPair.work.getPreferredEndpoint();
+      if (endpoint == null) {
+        currentUnassignedList.add(workPair);
+        continue;
+      }
+
+      FragIteratorWrapper iteratorWrapper = endpointIterators.get(endpoint);
+      if (iteratorWrapper == null) {
+        currentUnassignedList.add(workPair);
+        continue;
+      }
+
+      Integer assignment = iteratorWrapper.iter.next();
+      iteratorWrapper.count++;
```

Comment:
Here shouldn't we check if the "iteratorWrapper.count" is exceeding the "iteratorWrapper.maxCount" ?